easybuild/tools/module_generator.py

```diff
@@ -505,9 +506,15 @@ def load_module(self, mod_name, recursive_unload=False, unload_modules=None):
         body = []
         if unload_modules:
             body.extend([self.unload_module(m).strip() for m in unload_modules])
-        body.append(self.LOAD_TEMPLATE)
-
-        if build_option('recursive_mod_unload') or recursive_unload:
+        load_template = self.LOAD_TEMPLATE
+        # Lmod 7.6.1+ supports depends-on which does this most nicely:
+        if (build_option('recursive_mod_unload_depends_on') or
+            recursive_unload == 'depends_on') and modules_tool().has_depends_on:
```

Comment:
if `depends_on` is requested but not supported (`supports_depends_on` is `False`), then we should get an error rather than silently ignoring that request?

Reply:
Taking into account the above, this should become something like:

```python
load_template = self.LOAD_TEMPLATE
if use_depends_on:
    if modules_tool().supports_depends_on:
        load_template = self.LOAD_TEMPLATE_DEPENDS_ON
    else:
        raise EasyBuildError("depends_on statements in generated module are not supported by modules tool")